BotRoyaleUnity/Assets/Scripts/buildRobot.cs

```diff
@@ -7,57 +7,89 @@
-void Start()
-    {   
-        string jsonString = File.ReadAllText("Assets/Scripts/test.json");
-        var N = JSON.Parse(jsonString);
-        string test = N[0]["type"];
-        int a = 0;
-        while(N[a] != null){
-            pos = new Vector3((float)N[a]["x"],0.5f,(float)N[a]["y"]);
-            string type = N[a]["type"];
-            Quaternion rot;
-            string direcction = N[a]["direction"];
-            switch(direcction){
-                case "north":
-                    rot = Quaternion.Euler(90,0,0);
-                    break;
-                case "south":
-                    rot = Quaternion.Euler(-90,0,0);
-                    break;
-                case "east":
-                    rot = Quaternion.Euler(0,0,-90);
-                    break;
-                case "west":
-                    rot = Quaternion.Euler(0,0,90);
-                    break;
-                default:
-                    rot = Quaternion.Euler(0,0,0);
-                    break;
+    public GameObject block, center, spike;
+
+
+    // Start is called before the first frame update
+    void Start()
+    {
+        string jsonString = File.ReadAllText("Assets/Scripts/robot1.json");
+        GameObject robot1  = build(jsonString);
+        robot1.AddComponent<RoombaMovement>();
+        jsonString = File.ReadAllText("Assets/Scripts/robot2.json");
+        GameObject robot2  = build(jsonString);
+        robot2.AddComponent<RoombaMovement>();
+    }
+
+    void setParent(GameObject parent, GameObject child){
+        child.transform.parent = parent.transform;
+    }
+
+    GameObject build(string jsonString){
+        var json = JSON.Parse(jsonString);
+        int index = 0;
+        GameObject centerParent = null;
+        while (json[index] != null && centerParent == null)
+        {
+            if (json[index]["type"] == "center")
+            {
+                Vector3 pos = new Vector3((float)json[index]["x"], 0.5f, (float)json[index]["y"]);
+                Quaternion rot = Quaternion.Euler(90, 0, -90);
+                centerParent = Instantiate(center, pos, rot);
             }
-            switch(type){
-                case "block":
-                    Instantiate(block, pos, rot);  
-                    break;
-                case "center":
-                    Instantiate(center, pos, rot);
-                    break;
-                case "spike":
-                    Instantiate(spike, pos, rot);
-                    break;
+            index++;
+        }
+        index = 0;
+        while (json[index] != null)
+        {
+            string type = json[index]["type"];
+            if (type != "center")
+            {
+                GameObject childPart = null;
+                Vector3 pos = new Vector3((float)json[index]["x"], 0.5f, (float)json[index]["y"]);
+                Quaternion rot;
+                string direction = json[index]["direction"];
+                switch (direction)
```

Comment:
Not sure where to put this comment, but for semantics, it's good to have a gameobject be "facing" in the direction of the z axis, ie the blue arrow when the object is selected in the editor. 
<img width="596" alt="Screen Shot 2020-02-15 at 4 35 39 PM" src="https://user-images.githubusercontent.com/11395364/74597272-3c9fe780-5011-11ea-83ce-c04ba906c350.png">

Reply:
The roomba movement makes the object move in the y-axis, so I had to make the object face the y-axis

Reply:
Fair enough